runaction

```diff
@@ -48,11 +49,15 @@ readarray -d '' filepaths < <(find . -type f "${excludes[@]}" \
        ')'\
     \
     -print0)
-
+# remove files from filelist without shebang
+for file in "${!filepaths[@]}"; do
+    head -n1 "${filepaths[file]}" | grep -Eqs "$shebangregex" && continue
+    unset 'filepaths[file]'
+done
```

Comment:
This will not do anything, "filepaths" does not have any content until line 61.

This check does the same as the on line 60?

Reply:
It removes files that are matched by the `readarray` command in line 25 but do not contain the neccessary shebang line. From that line on the variable `filepaths` contains already the files that match in line 25 and the following ones.
When I run it for the branch and echo the content of `filepaths` in line 52, 58 and 65, I get the output:
```
$ bash runaction
./testfiles/bashfile.c ./testfiles/ignore/ignore.sh ./testfiles/test.bash ./testfiles/test.sh
./testfiles/ignore/ignore.sh ./testfiles/test.bash ./testfiles/test.sh
./testfiles/ignore/ignore.sh ./testfiles/test.bash ./testfiles/test.sh ./runaction ./testfiles/test
```
which seems correct to me since the loop above removes `./testfiles/bashfile.c` since it is a C file and not a bashscript. In the current master version this file would still be present in the list in the last line and in turn create failing tests for this file since it is not a valid bash script but C code. 

The check does the same as in line 60 but instead of adding when successful, it removes when unsuccessfull.

Reply:
You are right...
But that should not happen, that is a bug in the current code.
This line should not be there:

https://github.com/ludeeus/action-shellcheck/blob/c7d4e499f1a2fb59e837001884a4e05709b81eff/runaction#L29

Can you remove that and see if that gives you the expected result?
L43, L45-47 should also be removed.

Reply:
The file I had that problem in initially in one of my repositories () was `kshess.c` which resulted in:
```
In ./pyKSOPT/kshmul.c line 1:
/* kshmul.f -- translated by f2c (version 20100827).
^-- SC1127: Was this intended as a comment? Use # in sh.
^-- SC2148: Tips depend on target shell and yours is unknown. Add a shebang or a 'shell' directive.
^-- SC2211: This is a glob used as a command name. Was it supposed to be in ${..}, array, or is it missing quoting?
                                 ^-- SC1036: '(' is invalid here. Did you forget to escape it?
                                 ^-- SC1088: Parsing stopped here. Invalid use of parentheses?

In ./pyKSOPT/kshess.c line 1:
/* kshess.f -- translated by f2c (version 20100827).
^-- SC1127: Was this intended as a comment? Use # in sh.
^-- SC2148: Tips depend on target shell and yours is unknown. Add a shebang or a 'shell' directive.
^-- SC2211: This is a glob used as a command name. Was it supposed to be in ${..}, array, or is it missing quoting?
                                 ^-- SC1036: '(' is invalid here. Did you forget to escape it?
                                 ^-- SC1088: Parsing stopped here. Invalid use of parentheses?
```
So only removing L43, and 45-47 would resolve the problem for the example file I added to the repository (`bashfile.c`), but not resolve underlying problem. Line 29 and 31 would have to be removed as well.
The message from the commit 94453812848733dbcd8f35fd56e6a018abffc950, which initially introduced those filters, seems valid to me, but maybe the globbing is unneccessary and it would be better if it only did exact matching for the files named there:
* files with shbang lines that indicate a shell (sh, bash, ash, bsh, ksh, zsh)
  (_is done in the loop in L55 and can be kept_)
* typical home directory shell startup files:
  for posix sh & all shells: .profile
  for bash: .bashrc, .bash_aliases, .bash_completion, .bash_login, .bash_logout, .bash_profile
  for ksh: .profile
  for zsh: .zprofile, .zlogin, .zlogout, .zshenv, .zshrc
  (_is done in the `find` we are looking at by globbing and should be replaced by literal filenames maybe_)
* typical system-wide shell startup files:
  for posix sh & all shells: /etc/profile
  for bash: /etc/bashrc, /etc/bash_profile
  for ksh:
  for zsh: /etc/zprofile, /etc/zlogin, /etc/zlogout, /etc/zshenv, /etc/zshrc
  (_is done in the `find` we are looking at by globbing and should be replaced by literal filenames maybe_)
* files with suffixes: .bash, .zsh, .ksh, & .sh
  (_is done in the `find` we are looking at by globbing and can be kept_)

I have updated the code accordingly and removed the check for the shebang again.